src/coreComponents/constitutive/permeability/PressurePermeability.hpp

```diff
@@ -0,0 +1,256 @@
+                          m_maxPermeability,
+                          m_referencePermeability,
+                          m_permeability,
+                          m_dPerm_dPressure );
+  }
+
+  struct viewKeyStruct : public PermeabilityBase::viewKeyStruct
+  {
+    static constexpr char const * referencePermeabilityComponentsString() { return "referencePermeabilityComponents"; }
+    static constexpr char const * pressureDependenceConstantsString() { return "pressureDependenceConstants"; }
+    static constexpr char const * referencePressureString() { return "referencePressure"; }
+    static constexpr char const * referencePermeabilityString() { return "referencePermeability"; }
+    static constexpr char const * maxPermeabilityString() { return "maxPermeability"; }
+    static constexpr char const * pressureModelTypeString() { return "pressureModelType"; }
+  } viewKeys;
+
+  virtual void initializeState() const override final;
+
+private:
+
+  /// Permeability components at the reference pressure
+  R1Tensor m_referencePermeabilityComponents;
+
+  /// Pressure dependent coefficients for each permeability component
+  R1Tensor m_pressureDependenceConstants;
+
+  /// Reference pressure in the model
+  real64 m_referencePressure;
+
+  /// Maximum permeability 
+  real64 m_maxPermeability;
+
+  array3d< real64 > m_referencePermeability;
+
+  /// Pressure dependence model type
+  PressureModelType m_presModelType; 
+
+};
+
+GEOS_HOST_DEVICE
+GEOS_FORCE_INLINE
+void PressurePermeabilityUpdate::compute( real64 const & deltaPressure,
+                                          R1Tensor const pressureDependenceConstants,
+                                          real64 const (&referencePermeability)[3],
+                                          arraySlice1d< real64 > const & permeability,
+                                          arraySlice1d< real64 > const & dPerm_dPressure ) const
+{
+  for( localIndex i=0; i < permeability.size(); i++ )
+  {
+    real64 const perm = referencePermeability[i] * exp( pressureDependenceConstants[i] * deltaPressure );
+
+    permeability[i] = perm;
+    dPerm_dPressure[i] = perm * pressureDependenceConstants[i];
+  }
+}
+
+// 
+GEOS_HOST_DEVICE
+GEOS_FORCE_INLINE
+void PressurePermeabilityUpdate::compute( real64 const & deltaPressure,
+                                          R1Tensor const pressureDependenceConstants,
+                                          real64 const (&referencePermeability)[3],
+                                          real64 const maxPermeability,
+                                          arraySlice1d< real64 > const & permeability,
+                                          arraySlice1d< real64 > const & dPerm_dPressure ) const
+{
+  for( localIndex i=0; i < permeability.size(); i++ )
+  {
+    if( pressureDependenceConstants[i] < 1e-20 )
+    {
+      permeability[i] = referencePermeability[i]; 
+      dPerm_dPressure[i] = 0.0;
+    }
+    else 
```

Comment:
is this branching really necessary? if you want to avoid values that are too small we can add a check when reading the inputfile.

Reply:
> is this branching really necessary? if you want to avoid values that are too small we can add a check when reading the inputfile.

It is not a must for sure. I was thinking the component of `pressureDependenceConstants` can be set to zero, basically means no pressure dependence, but it will give us problem only when calculating the derivative for the hyperbolic type. Of course, I can add a check to avoid zero `pressureDependenceConstants`. So when using this model, one can assign a very small value if there is little pressure dependence in certain direction(s).

Reply:
yeah, I would just add a check in `postProcessInput`